### PR TITLE
compliance export: set start time to beg of day of end time

### PR DESCRIPTION
**Pull request by vjeffrey**
### :nut_and_bolt: Description
on the backend side of things, we are ignoring the start time for the
export endpoint, and simulating the behaviour here, where we say
start time is beg of day of end time. we would like to change this to
respect the provided start time for api users, so we need to adjust the
ui to send start time as beg of day of end time (instead of beg of day
from 10 days ago) so that we can maintain the behaviour of the ui export.

fixes https://github.com/chef/automate/issues/532

### :+1: Definition of Done
start time on export endpoint is beg of day of end time

### :athletic_shoe: Demo Script / Repro Steps
send some compliance scans in (chef_load_compliance_scans -N5)
rebuild ui
open dev tools
export json/csv
check headers

Screen Shot 2019-06-20 at 07 08 18

**this is not quite correct yet. it should have that Z at the end of the start time...help?**
^^ nevermind! got help from @scottopherson and all good now!

### CODE REVIEW

File: components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts [hunk 1]
```diff
@@ -173,13 +173,28 @@ export class StatsService {
...
+    filters.forEach(function (filt) {
+      if (filt['type'] === 'start_time') {
+        const val = moment(endtime['values'][0]).startOf('day');
+        filt['values'] = [val.format('YYYY-MM-DDTHH:mm:ssZ')];
+      }
+      filtersCopy.push(filt);
+    });
+
+    const body = { type: format,
+                   filters: filtersCopy};
```
[the remark this answers is not in the document]
Comment (vjeffrey) [reply components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts]: no, we called it above -- unless we change the logic around

File: components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts [hunk 2]
```diff
@@ -173,9 +173,17 @@ export class StatsService {
...
   downloadReport(format: string, filters: any[]): Observable<string> {
     const url = `${CC_API_URL}/reporting/export`;
-    const body = { type: format,
-                   filters: this.formatFilters(filters) };
+    // for export, we want to send the start_time as the beg of day of end time
+    // so we find the endtime in the filters, and then set start time to beg of that day
+    const endtime = filters.find(filter => filter['end_time']);
+    const filtersCopy = filters.map(filter => {
+      if (filter['start_time']) {
+        filter['start_time'] = moment(endtime).startOf('day');
```
Comment (lancewf) [inline components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts]: I think the below should add the 'Z' that you want, but do you want the time in UTC? 
```suggestion
       filter['start_time'] = moment(endtime).startOf('day').utc();
```

Comment (vjeffrey) [reply components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts]: ah, i should have edited the description - i got help from scott and the code is doing the right thing now

Comment (vjeffrey) [reply components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts]: the time gets set to utc during the format filters stuff

Comment (lancewf) [reply components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts]: Oh, ok. Well, I can test the code then.

**Pull request merged**